src/tracer_jaeger.rs

```diff
@@ -0,0 +1,24 @@
+use crate::{Reporter, Tracer};
+use rustracing::span::SpanReceiver;
+use rustracing_jaeger::span::SpanContextState;
+use std::thread;
+
+pub use rustracing::sampler::*;
+pub use rustracing_jaeger::{Result, Span as RtSpan, *};
+
+fn run_reporter_thread(service_name: &'static str, span_rx: SpanReceiver<SpanContextState>) {
+    thread::spawn(move || {
+        let reporter = Reporter::new(service_name).unwrap();
+        for span in span_rx {
+            reporter.report(&[span]).expect("could not report");
+        }
+    });
+}
+
+/// Create a Tracer that sends all spans to a jaeger reporter thread
+pub fn new_jaeger_tracer(service_name: &'static str) -> Tracer {
+    let (span_tx, span_rx) = crossbeam_channel::bounded(50);
+    run_reporter_thread(service_name, span_rx);
+    let tracer = Tracer::with_sender(AllSampler, span_tx);
+    tracer
+}
```

Comment:
I don't think this "specialized tracer" is a very good pattern. Reporter and Tracer are kept separate for a reason, and here we encourage users to forget about that separation. For instance, you might want to run several reporters on the span channel of a single Tracer. Or less commonly, send spans from multiple Tracers to a single Reporter.

I think that the awesome console tracing work you did should actually be implemented as a Reporter. Tracers are just for sending spans, Reporters are for rendering them, but this pattern combines the two concerns.

Reply:
If ConsoleReporter existed, then it would be easy to start a reporter thread that reports to both Jaeger and the console, using the same Tracer, which seems ideal